### ISSUE TITLE
Shared Drive

### ISSUE THREAD

**Issue by billbaird3** [ahochsteger/gmail2gdrive#53]
Hello!

Is there a way to specify a Shared Drive (instead of "My Drive") as the folder base?

Thanks!

**Comment by billbaird3** [comment]
Was able to get this working by modifying the following functions. I grabbed the "ID" by copying part of the URL for the folder I wanted to use (https://drive.google.com/drive/folders/xxxxxxxxxxxxxxxxx):

Inside "function getFolderByPath(path)":

 _//var folder = DriveApp.getRootFolder();
  folder = DriveApp.getFolderById('xxxxxxxxxxxxxxxxx');_

Inside "function getOrCreateFolder(folderName)":

    _//folder = getOrCreateSubFolder(DriveApp.getRootFolder(), folderArray);
    folder = getOrCreateSubFolder(DriveApp.getFolderById('xxxxxxxxxxxxxxxxx'), folderArray);_

**Comment by lightlySkewed** [comment]
Just a note. As of this writing, there's a try/ catch in "function getOrCreateFolder(folderName)" and the getFolderById will need to be added in both blocks. Ask me how I found out...

**Comment by itsbrex** [comment]
> 

How did you find out?

Also, I wasn't able to get it to work. Can you post the code you used to get it to work?

**Comment by lightlySkewed** [comment]
Bu dum, tsss...

Below is what worked for me.

```
function getOrCreateFolder(folderName) {
  var folder;
  try {
    //folder = getFolderByPath(folderName);
    folder = getOrCreateSubFolder(DriveApp.getFolderById('a_folder_id_string'), folderArray);
  } catch(e) {
    var folderArray = folderName.split("/");
    //folder = getOrCreateSubFolder(DriveApp.getRootFolder(), folderArray);
    folder = getOrCreateSubFolder(DriveApp.getFolderById('a_folder_id_string'), folderArray);
  }
  return folder;
}
```

**Comment by itsbrex** [comment]
Scratch that. Was able to get it to work removing the "/" in my folder name. 🙌  

Thank you both!

**Comment by lightlySkewed** [comment]
Rock & Roll!

**Comment by jgrunn** [comment]
@lightlySkewed, @itsbrex, 
I could not make the above work unfortunately, found a workaround though by saving it to mydrive then copy everything to the shared drive and delete the files inside the existing folder. However this is not optimal. , 

I have a few questions as I seem to miss something in your solution:
1.  do i need to apply both this:
` //var folder = DriveApp.getRootFolder();
  var folder = DriveApp.getFolderById('xxxxxxxxxxxxxxxxx');
`
and this?
`function getOrCreateFolder(folderName) {
  var folder;
  try {
    //folder = getFolderByPath(folderName);
    folder = getOrCreateSubFolder(DriveApp.getFolderById('a_folder_id_string'), folderArray);
  } catch(e) {
    var folderArray = folderName.split("/");
    //folder = getOrCreateSubFolder(DriveApp.getRootFolder(), folderArray);
    folder = getOrCreateSubFolder(DriveApp.getFolderById('a_folder_id_string'), folderArray);
  }
  return folder;
}`

2. I understand correctly the idea is to give the folderID of the root folder and then use the "folder": parameter within the rules?

**Comment by lightlySkewed** [comment]
Hi jgrunn,

Sorry it took me a minute to get back to you. I'm not 100% sure I understand your question. My answer right now is that, yes you need to tell your script where the root folder is (where to start) and then tell it where to build the new folders (and what to call them).

In my [code](https://github.com/lightlySkewed/gmail2gdrive/blob/29e9199f8823cfe9064a309e86ea41cc83cade7f/code.gs) these two steps are getOrCreateFolder() which defines the base/ root folder, and getOrCreateSubFolder() which builds the folder that will eventually house the files taken from Gdrive.

Bear in mind, it's been a few months since I looked at this so it's rusty. I also just noticed that I have a getFolderByPath() function that I don't even use. Sigh...

Best luck!

**Comment by ahochsteger** [comment]
Support for shared drives is now available through PR #72, maybe you want to give it a try.
Would be great to get some feedback, esp. if you have some edge cases that might have to be adressed.
